libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -30,29 +31,12 @@ VoteManager::~VoteManager() { daemon_->join(); }
 void VoteManager::setNetwork(std::weak_ptr<Network> network) { network_ = std::move(network); }
 
 void VoteManager::retreieveVotes_() {
-  LOG(log_si_) << "Retrieve unverified votes from DB";
-  auto unverified_votes = db_->getUnverifiedVotes();
-  for (auto const& v : unverified_votes) {
-    auto pbft_round = v->getRound();
-    auto hash = v->getHash();
-    {
-      UniqueLock lock(unverified_votes_access_);
-      if (unverified_votes_.count(pbft_round)) {
-        unverified_votes_[pbft_round][hash] = v;
-      } else {
-        std::unordered_map<vote_hash_t, std::shared_ptr<Vote>> votes{std::make_pair(hash, v)};
-        unverified_votes_[pbft_round] = std::move(votes);
-      }
-    }
-    LOG(log_dg_) << "Retrieved unverified vote " << *v;
-  }
-
   LOG(log_si_) << "Retrieve verified votes from DB";
   auto verified_votes = db_->getVerifiedVotes();
+  const auto pbft_step = db_->getPbftMgrField(PbftMgrRoundStep::PbftStep);
   for (auto const& v : verified_votes) {
     // Rebroadcast our own next votes in case we were partitioned...
-    if (v->getVoterAddr() == node_addr_ && v->getStep() >= FIRST_FINISH_STEP &&
-        db_->getPbftMgrField(PbftMgrRoundStep::PbftStep) > EXTENDED_PARTITION_STEPS) {
+    if (v->getStep() >= FIRST_FINISH_STEP && pbft_step > EXTENDED_PARTITION_STEPS) {
```

Comment:
Here only rebroadcast the node own next votes, not rebroadcast other peers votes. So need "v->getVoterAddr() == node_addr_"

Reply:
we are saving in DB only our own votes, so no need to check anymore here